### DISCUSSION TITLE
MET Unable to read NetCDF file

### DISCUSSION THREAD

**Discussion by craigr3** [dtcenter/METplus#1676] [File I/O] [answered]
I am trying to get MET to read an MPAS model file in netCDF format.  The header of the file is below:
netcdf CAFS_20220623T1800Z_fh06 {
dimensions:
        longitude = 1440 ;
        latitude = 721 ;
        level = 3 ;
        validtime = 6 ;
        cycletime = 6 ;
        time = 1 ;
variables:
        float longitude(longitude) ;
                longitude:units = "degrees_east" ;
        float latitude(latitude) ;
                latitude:units = "degrees_north" ;
        int level(level) ;
                level:units = "channel_number" ;
        int time(time) ;
        float Radiance(time, level, latitude, longitude) ;
                Radiance:description = "Radiance" ;
                Radiance:units = "W/m^2-sr" ;
        float TotalCloudCover(time, level, latitude, longitude) ;
                TotalCloudCover:description = "Total cloud cover" ;
                TotalCloudCover:units = "Percentage" ;
        int cycle_time(time, time, time, validtime) ;
                cycle_time:description = "Time array: (YYYY, MM, DD, HH, MM, SS)" ;
        int valid_time(time, time, time, validtime) ;
                valid_time:description = "Time array: (YYYY, MM, DD, HH, MM, SS)" ;
data:

I want to verify the TotalCloudCover variable.    I tried the following in my MET gridstat conifg file:
Fcst = {
     lead_time = "24";
    field = [
                  {name = "TotalCloudCover"; level = "(0,0,*,*)"; }

I have also tried for level "(*,*)" and "(0,*,*)".
There are asterisks between the commas but they don't show up here for some reason.
What level parameter should I be using.  When MET runs it gives the Trouble reading file error.

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi @craigr3, I see you're having trouble about reading gridded NetCDF data into MET.

I'd definitely start by running this file through the plot_data_plane utility to make sure that MET can read it.

First, let me offer a tip about formatting comments in GitHub discussions. You can include embedded literal strings by enclosing them in single back-ticks, like this... `level = "(*,*)";`
Or you can format multi-line literals by enclosing them with 3 back-ticks, like this:
```
netcdf CAFS_20220623T1800Z_fh06 {
dimensions:
longitude = 1440 ;
latitude = 721 ;
level = 3 ;
validtime = 6 ;
cycletime = 6 ;
time = 1 ;
```
If you enclose asterisks in single back-ticks like this... `*`, then they'll appear in discussions.
As for the actual issue...

My guess is that this data really is on a regularly-spaced lat/lon grid.
But I'm guessing it lacks the global `conventions = "CF-X.Y";` attribute which would identify it as being a NetCDF file following the Climate-Forecast convention. Try running:
```
ncdump -h CAFS_20220623T1800Z_fh06.nc | egrep -i convention
```
If the convention is not specified, I'd recommend trying to manually define it using the `file_type` option, like this:
```
plot_data_plane CAFS_20220623T1800Z_fh06.nc TotalCloudCover.ps 'name="TotalCloudCover"; level="(0,0,*,*)"; file_type=NETCDF_NCCF;'
```
Does that produce any better results?

**Reply by JohnHalleyGotway** [reply]
@craigr3 just wanted to follow up on this discussion from a while back. Are you still having issues related to the CAFS data?

If not, please feel free to select the **Mark as answer** button to select the best answer for the issue.

Thanks!
